File: Kepler-Server/src/main/java/org/alexdev/kepler/game/catalogue/CataloguePackage.java
package org.alexdev.kepler.game.catalogue;

import org.alexdev.kepler.game.item.base.ItemDefinition;
import org.alexdev.kepler.game.item.ItemManager;

public class CataloguePackage {
    private String saleCode;
    private int definitionId;
    private int specialSpriteId;
    private int amount;

    public CataloguePackage(String saleCode, int definitionId, int specialSpriteId, int amount) {
        this.saleCode = saleCode;
        this.definitionId = definitionId;
        this.specialSpriteId = specialSpriteId;
        this.amount = amount;
    }

    public String getSaleCode() {
        return saleCode;
    }

    public int getDefinitionId() {
        return definitionId;
    }

    public ItemDefinition getDefinition() {
        return ItemManager.getInstance().getDefinition(this.definitionId);
    }

    public int getSpecialSpriteId() {
        return specialSpriteId;
    }

    public int getAmount() {
        return amount;
    }
}

File: src/game/catalogue/catalogue_package.c
#include <stdlib.h>
#include <stdio.h>

#include "log.h"

#include "game/items/item_manager.h"
#include "catalogue_package.h"

catalogue_package *catalogue_package_create(char *salecode, int definition_id, int special_sprite_id, int amount) {
    catalogue_package *package = malloc(sizeof(catalogue_package));
    package->sale_code = salecode;
    package->definition_id = definition_id;
    package->special_sprite_id = special_sprite_id;
    package->amount =  amount;
    package->definition = item_manager_get_definition_by_id(definition_id);

    if (package->definition == NULL) {
        log_warn("Warning! The package %s has no valid definition", package->sale_code);
    }

    return package;
}

/**
 * Dispose catalogue package.
 *
 * @param page the catalogue package to dispose
 */
void catalogue_package_dispose(catalogue_package *package) {
    free(package->sale_code);
    free(package);
}